Added basic integration for the cancel feature.

skazi0 opened the pull request:
This is useful for testing the begin upgrade without a need for restoring previous cloud state.

skazi0 committed:
upgrade: Added basic integration for the cancel feature.

Itxaka reviewed (approved):
Looking good!

santiph commented on assets/app/widgets/crowbar-upgrade-nav/crowbar-upgrade-nav.directive.js:
```diff
@@ -11,7 +11,23 @@
+
+    CrowbarUpgradeNavController.$inject = ['$state', 'upgradeFactory'];
+
+    function CrowbarUpgradeNavController($state, upgradeFactory) {
```
Widgets, as they've been conceived, are meant to be reusable components across different projects in the future simply by adding the module as a dependency. Therefore, they should not depend on other modules, like `suseData` nor on the features modules to determine the states and consume the API.

Please, provide those attributes through the controller and their scopes.

skazi0 committed:
upgrade: Removed dependency inside directive

Callback attached to 'Cancel' button has been moved outside of the directive code.
Now it is passed from outside (controller) using `on-cancel` attribute.

Itxaka reviewed (approved):
Still looking good to me

skazi0 commented:
rebased

santiph commented on assets/app/widgets/crowbar-upgrade-nav/crowbar-upgrade-nav.directive.jade:
```diff
@@ -1,4 +1,4 @@
-button.btn.btn-default.pull-left(type='button')
+button.btn.btn-default.pull-left(type='button', ng-click="onCancel()")
   span(aria-hidden="true")
   | Cancel Upgrade
```
translation needed

santiph commented on assets/app/features/upgrade/upgrade.controller.js:
```diff
@@ -47,5 +49,14 @@
+         */
+        function cancelUpgrade() {
+            upgradeFactory.cancelUpgrade().then(function(/* response */) {
+                $state.go('upgrade-landing');
```
After cancelation, user needs to be redirected to crowbar dashboard.
Completely out of the scope of this PR, but I'd like to ask you to add a comment in the code and https://trello.com/c/aP3CKeqb/28-2-17-cancellation-routine-page-creation to not lose track of it

skazi0 replied:
Comment added in code. I don't have write access to this trello so can't do anything there.

santiph commented on assets/app/data/crowbar/services/upgrade.factory.spec.js:
```diff
@@ -153,6 +154,32 @@ describe('Upgrade Factory', function () {
+                });
+                $httpBackend.flush();
+            });
+        });
```
When executed, it's supposed to also redirect the user to crowbar dashboard... Again, out of the scope, but I'd appreciate a comment in the code and the trelo card: https://trello.com/c/aP3CKeqb/28-2-17-cancellation-routine-page-creation

skazi0 replied:
The redirect should be done in the controller so the service doesn't need any changes.

santiph reviewed (changes requested):
Good implementation!
Please address the minor request so we don't lose track of them and consider this approved.

skazi0 committed:
upgrade: Moved navbar strings to translations file

Itxaka commented:
![](https://github.trello.services/images/mini-trello-icon.png) [(3) S51P10 (S50P9 S49P20) #6 Layout - Navigation bar](https://trello.com/c/LNr710jO/104-3-s51p10-s50p9-s49p20-6-layout-navigation-bar)

santiph commented:
Approved as a temporary solution to be properly implemented (or completely removed, depending on the PO decision) in a future sprint
